src/NLog/Config/LoggingRuleLevelFilter.cs

```diff
@@ -0,0 +1,251 @@
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+namespace NLog.Config
+{
+    using System;
+    using System.Collections.Generic;
+    using NLog.Common;
+    using NLog.Internal;
+    using NLog.Layouts;
+
+    internal interface ILoggingRuleLevelFilter
+    {
+        /// <summary>
+        /// Level enabled flags for each LogLevel ordinal
+        /// </summary>
+        bool[] LogLevels { get; }
+        /// <summary>
+        /// Reconfigure existing level configuration with the specified config
+        /// </summary>
+        /// <returns>New filter</returns>
+        ILoggingRuleLevelFilter SetLoggingLevels(LogLevel minLevel, LogLevel maxLevel, bool enable);
+    }
+
+    /// <summary>
+    /// Default filtering with static level config
+    /// </summary>
+    class LoggingRuleLevelFilter : ILoggingRuleLevelFilter
+    {
+        public static readonly LoggingRuleLevelFilter Off = new LoggingRuleLevelFilter();
+        public bool[] LogLevels { get; }
+
+        public LoggingRuleLevelFilter(bool[] logLevels = null)
+        {
+            LogLevels = new bool[LogLevel.MaxLevel.Ordinal + 1];
+            if (logLevels != null)
+            {
+                for (int i = 0; i < Math.Min(logLevels.Length, LogLevels.Length); ++i)
+                    LogLevels[i] = logLevels[i];
+            }
+        }
+
+        public ILoggingRuleLevelFilter SetLoggingLevels(LogLevel minLevel, LogLevel maxLevel, bool enable)
+        {
+            if (!ReferenceEquals(LogLevels, Off.LogLevels))
+            {
+                for (int i = minLevel.Ordinal; i <= maxLevel.Ordinal; ++i)
+                    LogLevels[i] = enable;
+                return this;
+            }
+
+            return new LoggingRuleLevelFilter().SetLoggingLevels(minLevel, maxLevel, enable);
```

Comment:
this is a bit unexpected. Why the `new LoggingRuleLevelFilter()`. The method is now a combination of "set" and "create"

Reply:
Because it is difficult to combine dynamic filters with hardcoded assignment of levels. So it converts dynamic-filter to simple-filter if needed. If already simple-filter then it just modifies and returns this.

Reply:
Have now changed it into `GetSimpleFilterForUpdate()` that returns `LoggingRuleLevelFilter` (that has the method `SetLoggingLevels`)